### ISSUE TITLE
US3 - 4 Implement a training pipeline around the model

### ISSUE THREAD

**Issue by desireeschoonbrood** [ADS2025-A2/team-A2-recsys#69]
### Objective

Create a standard script to train a model end-to-end.

### Assignee (GitHub handle)

desireeschoonbrood

### Estimate (hours)

3

### Checklist

- Loads raw data and splits into train, val, test via Task 2.
- Instantiates the baseline Spotlight model from Task 3
- Trains the model for configurable number of epochs.
- It logs at least: per-epoch training loss, a validation metric
- On successful completion, the training script saves the trained model
